### PR TITLE
frida-asan: Add windows x64 support

### PR DESCRIPTION
**Pull request by s1341**
Add's initial frida-ASAN support for windows x64 targets, and enables ASAN in the `frida_gdiplus` example fuzzer.

### CODE REVIEW

**Comment by s1341** [comment]
Note that this currently requires the following PRS:
- https://github.com/xacrimon/dashmap/pull/267
- https://github.com/StephanvanSchaik/mmap-rs/pull/12
- https://github.com/frida/frida-rust/pull/96

**Comment by s1341** [comment]
@tokatoka yeah. I don't really see a reason for that function to be pre-assembled anymore.

File: libafl_frida/src/asan/asan_rt.rs [hunk 1]
```diff
@@ -2648,7 +2647,7 @@ impl AsanRuntime {
                 }
                 X86Register::Rdi => {
                     // In this case rdi is already clobbered, so we want it from the stack (we pushed rdi onto stack before!)
-                    writer.put_mov_reg_reg_offset_ptr(X86Register::Rsi, X86Register::Rsp, -0x28);
+                    writer.put_mov_reg_reg_offset_ptr(X86Register::Rsi, X86Register::Rsp, -0x18);
```
Comment (tokatoka) [inline libafl_frida/src/asan/asan_rt.rs:2244]: hm.. why? `rdi` is on `stack-0x28`

Comment (s1341) [reply libafl_frida/src/asan/asan_rt.rs:2244]: https://github.com/AFLplusplus/LibAFL/pull/1314

File: libafl_frida/src/alloc.rs [hunk 2]
```diff
@@ -532,6 +498,49 @@ impl Allocator {
         map_to_shadow!(self, start)
     }
 
+    /// Checks whether the given address up till size is valid unpoisoned shadow memory.
+    #[inline]
+    #[must_use]
+    pub fn check_shadow(&self, address: *const c_void, size: usize) -> bool {
```
Comment (WilliamParks) [inline libafl_frida/src/alloc.rs:504]: When I was trying to debug some other issues, I used this function, vice the assembly one, as it was easier to debug.  I had issues with some of the corner cases, but am not sure if my fixes were correct.

- If address isn't aligned to 16 bits, and size < 8, shadow_size underflows.
- I think the checks for shadow_addrs need to be inverted?  For example, if there are 4 bytes at the end of an allocation, unpoison fills them with 0xf0, while the checks here compare that with 0x04 (on lines 514 and 535)

**Comment by domenukk** [comment]
Any reason not to merge this? Doesn't need to be perfect from the beginning,  anything that runs is better than nothing?

**Comment by s1341** [comment]
It can probably be merged. I'm not on this actively at the moment, but will get back to it soon hopefully.

**Comment by tokatoka** [comment]
we need to look at this before merge
https://github.com/AFLplusplus/LibAFL/pull/1298#discussion_r1233980242

**Comment by domenukk** [comment]
We can merge this now that frida got updated :)

**Comment by domenukk** [comment]
Any chance to still merge this? :D
Would be great to have, cc @s1341

**Comment by s1341** [comment]
I'm slowly going to ty to merge pieces of this.

**Comment by domenukk** [comment]
Continuing work in #1607

**Pull request closed without merging**